### PR TITLE
Add message request support

### PR DESCRIPTION
**Pull request by AllanWang**
Full credits to [QuickUnit's commit](https://github.com/QuickUnit/facebook-chat-api/commit/f6828953594ca6b26810e363161fe2b210943be3)

I changed the description a bit and changed the type checks as mentioned [here ](https://github.com/Schmavery/facebook-chat-api/pull/247#issuecomment-233505141).

I tested it and it seems to be working.

I did not modify the getThreadInfo file. The indents just changed.

### CODE REVIEW

File: src/getThreadInfo.js [hunk 1]
```diff
@@ -4,49 +4,58 @@ var utils = require("../utils");
 var log = require("npmlog");
 
 module.exports = function(defaultFuncs, api, ctx) {
-  return function getThreadInfo(threadID, callback) {
```
Comment (bsansouci) [inline src/getThreadInfo.js]: Can you re-indent this back to 2 spaces?

File: src/getThreadList.js [hunk 2]
```diff
@@ -4,40 +4,54 @@ var utils = require("../utils");
 var log = require("npmlog");
 
 module.exports = function(defaultFuncs, api, ctx) {
-  return function getThreadList(start, end, callback) {
```
Comment (bsansouci) [inline src/getThreadList.js]: re-indent to 2 spaces

File: src/getThreadList.js [hunk 3]
```diff
@@ -4,40 +4,54 @@ var utils = require("../utils");
...
+            } else if (utils.getType(pending) !== 'Boolean') {
+                throw {
+                    error: "please pass a Boolean as a third argument."
+                };
+            } else {
+                throw {
+                    error: "getThreadList: need callback"
+                };
+            }
+        }
```
Comment (bsansouci) [inline src/getThreadList.js]: Here you can add 
```js
else if (utils.getType(pending) != "Boolean") {
  throw {
    error: "Please pass a boolean as a third argument."
  };
}
```

File: src/getThreadList.js [hunk 4]
```diff
@@ -4,40 +4,54 @@ var utils = require("../utils");
...
 module.exports = function(defaultFuncs, api, ctx) {
-  return function getThreadList(start, end, callback) {
-    if(!callback && utils.getType(end) !== 'Number') {
-      throw {error: "please pass an number as a second argument."};
-    }
+    return function getThreadList(start, end, pending, callback) {
+        if (utils.getType(callback) === 'Undefined') {
+            if (utils.getType(end) !== 'Number') {
+                throw {
+                    error: "please pass a number as a second argument."
```
Comment (bsansouci) [inline src/getThreadList.js]: `Please` capital letter.

File: src/getThreadList.js [hunk 5]
```diff
@@ -4,40 +4,54 @@ var utils = require("../utils");
...
+                };
+            } else if (utils.getType(pending) === 'Function') {
+                callback = pending;
+                pending = false; //default to inbox
+            } else if (utils.getType(pending) !== 'Boolean') {
+                throw {
+                    error: "please pass a Boolean as a third argument."
```
Comment (bsansouci) [inline src/getThreadList.js]: `Please` capital letter.

**Comment by bsansouci** [comment]
Looks good. Ready to merge when you fix the nitpicking. Thanks a lot!

**Comment by AllanWang** [comment]
@bsansouci 

```javascript
if (utils.getType(accept) !== 'Boolean') {
            throw {
                error: "Please pass a boolean as a second argument."
            };
        }
```

Should that be != instead of !== as well?

**Comment by bsansouci** [comment]
Oh no it should always be `!==` and `===` _except_ when you're comparing with `null`. Then you'd want to use `!=` and `==`. So in this case use `!==`.

**Comment by AllanWang** [comment]
Can you check now?

I also kept it as 

```javascript
      } else if (utils.getType(pending) != 'Boolean') {
        throw {
          error: "Please pass a Boolean as a third argument."
        };
      } else {
        throw {
          error: "getThreadList: need callback"
        };
      }
```
Since you'll need to throw the last callback error if there is no boolean and no callback

**Comment by bsansouci** [comment]
Sounds good. Thanks.

**Pull request merged**